### ISSUE TITLE
Ubuntu 23.04 (Lunar Lobster) PPA non functional

### ISSUE THREAD

**Issue by SegaGameBoyStation** [xemu-project/xemu#1445]
### Bug Description

Ubuntu Launar Lobster PPA has no usable packages

### Expected Behavior

adding PPA should allow for installing packages

### xemu Version

xemu - 0.7.90-0~202305161702~ubuntu23.04.1 

### System Information

Ubuntu 23.04

### Additional Context

all builds for Lunar Lobster on the PPA have failed so there is nothing that can be installed

**Comment by Mucephie** [comment]
I found that my base install of 23.04 was having trouble compiling `xemu` from the source and needed an `apt install` of  `libglu1-mes-dev` to get past one build issue. But even with this library installed, the compilation failed at a later point with a much less clear reason. Here is the relevant terminal output 
```
[1384/1385] Compiling C object qemu-system-i386.p/softmmu_main.c.o
../softmmu/main.c:31:5: warning: no previous prototype for ‘qemu_main’ [-Wmissing-prototypes]
   31 | int qemu_main(int argc, char **argv, char **envp)
      |     ^~~~~~~~~
[1385/1385] Linking target qemu-system-i386
In function ‘stbi__parse_uncompressed_block’,
    inlined from ‘stbi__parse_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4415:15,
    inlined from ‘stbi__do_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4439:11,
    inlined from ‘stbi_zlib_decode_malloc_guesssize_headerflag’ at ../ui/thirdparty/stb_image/stb_image.h:4470:8,
    inlined from ‘stbi__parse_png_file.constprop’ at ../ui/thirdparty/stb_image/stb_image.h:5143:39:
../ui/thirdparty/stb_image/stb_image.h:4348:19: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 4348 |       header[k++] = stbi__zget8(a);
      |                   ^
../ui/thirdparty/stb_image/stb_image.h: In function ‘stbi__parse_png_file.constprop’:
../ui/thirdparty/stb_image/stb_image.h:4334:12: note: at offset [5, 9] into destination object ‘header’ of size 4
 4334 |    stbi_uc header[4];
      |            ^
In function ‘stbi__parse_uncompressed_block’,
    inlined from ‘stbi__parse_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4415:15,
    inlined from ‘stbi__do_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4439:11,
    inlined from ‘stbi_zlib_decode_malloc_guesssize_headerflag’ at ../ui/thirdparty/stb_image/stb_image.h:4470:8,
    inlined from ‘stbi__parse_png_file.constprop’ at ../ui/thirdparty/stb_image/stb_image.h:5143:39:
../ui/thirdparty/stb_image/stb_image.h:4348:19: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 4348 |       header[k++] = stbi__zget8(a);
      |                   ^
../ui/thirdparty/stb_image/stb_image.h: In function ‘stbi__parse_png_file.constprop’:
../ui/thirdparty/stb_image/stb_image.h:4334:12: note: at offset [5, 9] into destination object ‘header’ of size 4
 4334 |    stbi_uc header[4];
      |            ^
In function ‘stbi__parse_uncompressed_block’,
    inlined from ‘stbi__parse_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4415:15,
    inlined from ‘stbi__do_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4439:11,
    inlined from ‘stbi_zlib_decode_malloc_guesssize_headerflag’ at ../ui/thirdparty/stb_image/stb_image.h:4470:8,
    inlined from ‘stbi__parse_png_file.constprop’ at ../ui/thirdparty/stb_image/stb_image.h:5143:39:
../ui/thirdparty/stb_image/stb_image.h:4348:19: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 4348 |       header[k++] = stbi__zget8(a);
      |                   ^
../ui/thirdparty/stb_image/stb_image.h: In function ‘stbi__parse_png_file.constprop’:
../ui/thirdparty/stb_image/stb_image.h:4334:12: note: at offset [6, 10] into destination object ‘header’ of size 4
 4334 |    stbi_uc header[4];
      |            ^
In function ‘stbi__parse_uncompressed_block’,
    inlined from ‘stbi__parse_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4415:15,
    inlined from ‘stbi__do_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4439:11,
    inlined from ‘stbi_zlib_decode_malloc_guesssize_headerflag’ at ../ui/thirdparty/stb_image/stb_image.h:4470:8,
    inlined from ‘stbi__parse_png_file.constprop’ at ../ui/thirdparty/stb_image/stb_image.h:5143:39:
../ui/thirdparty/stb_image/stb_image.h:4348:19: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 4348 |       header[k++] = stbi__zget8(a);
      |                   ^
../ui/thirdparty/stb_image/stb_image.h: In function ‘stbi__parse_png_file.constprop’:
../ui/thirdparty/stb_image/stb_image.h:4334:12: note: at offset [5, 9] into destination object ‘header’ of size 4
 4334 |    stbi_uc header[4];
      |            ^
In function ‘stbi__parse_uncompressed_block’,
    inlined from ‘stbi__parse_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4415:15,
    inlined from ‘stbi__do_zlib’ at ../ui/thirdparty/stb_image/stb_image.h:4439:11,
    inlined from ‘stbi_zlib_decode_malloc_guesssize_headerflag’ at ../ui/thirdparty/stb_image/stb_image.h:4470:8,
    inlined from ‘stbi__parse_png_file.constprop’ at ../ui/thirdparty/stb_image/stb_image.h:5143:39:
../ui/thirdparty/stb_image/stb_image.h:4348:19: warning: writing 1 byte into a region of size 0 [-Wstringop-overflow=]
 4348 |       header[k++] = stbi__zget8(a);
      |                   ^
../ui/thirdparty/stb_image/stb_image.h: In function ‘stbi__parse_png_file.constprop’:
../ui/thirdparty/stb_image/stb_image.h:4334:12: note: at offset [5, 12] into destination object ‘header’ of size 4
 4334 |    stbi_uc header[4];
      |            ^
In function ‘qemu_extend_irqs’,
    inlined from ‘qemu_allocate_irqs’ at ../hw/core/irq.c:66:12,
    inlined from ‘ahci_realize’ at ../hw/ide/ahci.c:1547:12:
../hw/core/irq.c:57:51: warning: argument 1 range [18446744071562067968, 18446744073709551615] exceeds maximum object size 9223372036854775807 [-Walloc-size-larger-than=]
   57 |     s = old ? g_renew(qemu_irq, old, n + n_old) : g_new(qemu_irq, n);
      |                                                   ^
/usr/include/glib-2.0/glib/gmem.h: In function ‘ahci_realize’:
/usr/include/glib-2.0/glib/gmem.h:99:10: note: in a call to allocation function ‘g_malloc_n’ declared here
   99 | gpointer g_malloc_n       (gsize         n_blocks,
      |          ^


real	4m42.023s
user	15m18.851s
sys	1m49.936s
+ package_linux
+ rm -rf dist
+ mkdir -p dist
+ cp build/qemu-system-i386 dist/xemu
+ python3 ./scripts/gen-license.py
```
There seems to be two issues here, one on step [1384/1385] with `qemu_main` and another at step [1385/1385] with some longer network of issues that also seems to involve `qemu`. I was compiling the latest `xemu-0.7.96` on `Ubuntu 23.04 Lunar Lobster`  following the 'build from source' instructions in the documentation.

**Comment by antangelo** [comment]
The build logs you provide are successful, the issues shown are only warnings and do not fail the build.

On a 22.04 docker container following the build guide on the website, `libglu1-mesa-dev` is automatically installed as part of the dependencies, whereas on a 23.04 container it is not. We may just need to require it as an explicit dependency.

**Comment by StianOby** [comment]
This can be closed as resolved, since builds for 23.04 have been available for quite some time: https://launchpad.net/~mborgerson/+archive/ubuntu/xemu?field.series_filter=lunar 

However, 23.10 is lacking builds, but I have created a separate issue for that (see #1554 )
